low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -231,19 +246,24 @@ def properties_changed_handler(self, interface, changed_properties,
         if self.state != self.previous_state:
             if self.state == "active":
                 if self._service_state is FailedState:
+                    self.waiting_timestamp = time.time()
+                    Service.active_waiting_services.add(self.name)
```

Comment:
Hi, so Instead of keeping the actions here in if-else condition, what if we create a `ActiveWaiting` & `FailedWaiting` state classes and add these 2 lines in `enter()` method of those classes, then all we have to do will be changing the state to ActiveWaiting here, and in our main loop we will keep checking if waiting time is over to switch to state to Active.

Reply:
*_After adding these states we will again have to calculate transitions and verify actions so **I am good with the current changes**._